src/util/json_stream.h

```diff
@@ -0,0 +1,151 @@
+/*******************************************************************\
+
+Module:
+
+Author: Peter Schrammel
+
+\*******************************************************************/
+
+
+#ifndef CPROVER_UTIL_JSON_STREAM_H
+#define CPROVER_UTIL_JSON_STREAM_H
+
+#include "json.h"
+#include "invariant.h"
+
+class json_stream_objectt;
+class json_stream_arrayt;
+
+/// New class whose aim is to stream newly created JSON
+/// objects directly instead of waiting for the object
+/// to be fully formed in memory and then print it.
+/// 
+/// This approach was necessary to fix some issues with
+/// CBMC running out of memory when it had to print some
+/// huge traces for some programs.
+class json_streamt
```

Comment:
The only public operation is `close` - that's truly weird. Please create a meaningful API, which might have `virtual ...(...) = 0;` members - but at least it would explain how to use this thingy.

Reply:
The only public facility that this base class provides is close().
output_current and output_finalizer should be made abstract.

Reply:
I don't think this has happened? Maybe it has, but they are all `protected` so I have no idea what is the public interface of (children of) `json_streamt`.